modules/store/spec/feature_creator.spec.ts

```diff
@@ -98,6 +104,104 @@ describe('createFeature()', () => {
     });
   });
 
+  describe('derived selectors', () => {
+    it('should create a feature', () => {
+      const initialState = { count1: 9, count2: 10 };
+      const counterFeature = createFeature({
+        name: 'counter',
+        reducer: createReducer(initialState),
+        derivedSelectors: ({
+          selectCounterState,
+          selectCount1,
+          selectCount2,
+        }) => ({
+          selectSquaredCount2: createSelector(
+            selectCounterState,
+            ({ count2 }) => count2 * count2
+          ),
+          selectTotalCount: createSelector(
+            selectCount1,
+            selectCount2,
+            (count1, count2) => count1 + count2
+          ),
+        }),
+      });
+
+      expect(counterFeature.selectCounterState({ counter: initialState })).toBe(
+        initialState
+      );
+      expect(counterFeature.selectCount1({ counter: initialState })).toBe(
+        initialState.count1
+      );
+      expect(counterFeature.selectCount2({ counter: initialState })).toBe(
+        initialState.count2
+      );
+      expect(
+        counterFeature.selectSquaredCount2({ counter: initialState })
+      ).toBe(initialState.count2 * initialState.count2);
+      expect(counterFeature.selectTotalCount({ counter: initialState })).toBe(
+        initialState.count1 + initialState.count2
+      );
+      expect(Object.keys(counterFeature)).toEqual([
+        'name',
+        'reducer',
+        'selectCounterState',
+        'selectCount1',
+        'selectCount2',
+        'selectSquaredCount2',
+        'selectTotalCount',
+      ]);
+    });
+
+    it('should override base selectors if derived selectors have the same names', () => {
```

Comment:
What would be a good use case of this?

I'm asking because I don't really like this 😅
Personally, if someone does this this could lead to some frustrations on my end because I wouldn't expect it. As far as I see, there would also be no way to select the state value.

Reply:
This can be useful when adding entity selectors to the feature object.

When `createFeature` is used with `EntityState`, it will generate `selectEntities` and `selectIds` selectors. With current behavior, we are able to add derived entity selectors in the following way:

```ts
export const productsFeature = createFeature({
  name: 'products',
  reducer: createReducer(adapter.getInitialState()),
  extraSelectors: ({ selectProductsState }) => ({
    ...adapter.getSelectors(selectProductsState),
  }),
});
```

In this example, `selectEntities` and `selectIds` returned by `adapter.getSelectors` will override `selectEntities` and `selectIds` generated by `createFeature`. However, if we don't allow this behavior, then `adapter.getSelectors` needs to be used in the following way to avoid compilation error:

```ts
export const productsFeature = createFeature({
  name: 'products',
  reducer: createReducer(adapter.getInitialState()),
  extraSelectors: ({ selectProductsState }) => {
    const { selectAll, selectTotal } = adapter.getSelectors(selectProductsState);
    return { selectAll, selectTotal };
  },
});
```

Reply:
Oh I see, I didn't think of that.
Thanks Marko!